lib/src/controls/codec.rs

```diff
@@ -70,3 +71,294 @@ impl ExtControlTrait for Vp8Frame {
     const ID: u32 = bindings::V4L2_CID_STATELESS_VP8_FRAME;
     type PAYLOAD = v4l2_ctrl_vp8_frame;
 }
+
+/// Safe wrapper over [`v4l2r::bindings::V4L2_CID_MPEG_VIDEO_HEADER_MODE`]
+#[repr(i32)]
+#[derive(N, Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum VideoHeaderMode {
+    Separate = bindings::v4l2_mpeg_video_header_mode_V4L2_MPEG_VIDEO_HEADER_MODE_SEPARATE as i32,
+    JoinedWith1stFrame =
+        bindings::v4l2_mpeg_video_header_mode_V4L2_MPEG_VIDEO_HEADER_MODE_JOINED_WITH_1ST_FRAME
+            as i32,
+}
+
+impl ExtControlTrait for VideoHeaderMode {
+    const ID: u32 = bindings::V4L2_CID_MPEG_VIDEO_HEADER_MODE;
+    type PAYLOAD = i32;
+}
+
+impl From<VideoHeaderMode> for i32 {
```

Comment:
We should be able to skip these `From` implementations, as the `repr(i32)` makes it possible to just apply `as i32` to instances of the type. Or is there another reason to have them?

Reply:
It makes function like [this](https://github.com/chromeos/cros-codecs/blob/709c5d5545ab8b68360d787eba51f174074745a6/src/backend/v4l2/encoder.rs#L393) possible. AFAIK there is no way of ensuring that generic argument can be casted using `as i32`. Without the `From` that function would have to accept `i32` and `const ID` in order to create `SafeExtControl` and compare the values which IMHO defeats the purpose of `ExtControlTrait`.

Reply:
Ah, I see. Let's keep it then. :+1: